compiler/rustc_parse/src/parser/item.rs

```diff
@@ -588,15 +588,8 @@ impl<'a> Parser<'a> {
         let has_for = self.eat_keyword(kw::For);
         let missing_for_span = self.prev_token.span.between(self.token.span);
 
-        let ty_second = if self.token == token::DotDot {
-            // We need to report this error after `cfg` expansion for compatibility reasons
```

Comment:
This suggests that we intend the following (historically wrongly gated 😭) to pass:
```rust
#[cfg(dont)]
impl Trait for .. {}
```
given that this should have emitted a future compat warning (from a quick test it doesn't seem to do that though????, I might be doing it wrong though)

but this is a breaking change (a probably justified one, but one)

maybe this should just use some dummy node in the meantime to not block this